### PR TITLE
[Feature] Telescopic Baton

### PR DESCRIPTION
**Pull request by Remuchi**
# Description

Introduces a telescopic baton - a new self-defense weapon for command staff

DEVELOPERS:
* Adds a way to apply status effects with pre-made component.

AWAITING DISCUSSION:
* Currently, the system works that players are rewarded for their skill and ability to click precisely. The batton will knockdown someone only in first 1.5 seconds after extending it. If you fail to do so, you'll have to retract it and extend once again, which takes about 4 seconds total. 

---
<details><summary><h1>Media</h1></summary>
<p>

https://github.com/user-attachments/assets/cc19bf44-391e-4d27-9be9-96d3d77bebc8

</p>
</details>

---

# Changelog

:cl:
- add: Added telescopic baton - a self-defense weapon for Command staff.

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/telescopic_baton.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    soundDeactivate:
+      path: /Audio/Weapons/telescopicoff.ogg
+      params:
+        volume: -2
+  - type: ItemToggleDisarmMalus
+    activatedDisarmMalus: 0.6
+  - type: ItemToggleMeleeWeapon
+    activatedDamage:
+      types:
+        Blunt: 12
```
Comment (FoxxoTrystan) [inline Resources/Prototypes/Entities/Objects/Weapons/Melee/telescopic_baton.yml:28]: Sad i enjoyed my Baton doing no damage but only knock down so its gave them time to run.

Personnaly il prefer it to be only knock down or minimal damage near 1-5, making them avoid using in combat and more as a defense tool only.

**Review by FoxxoTrystan — APPROVED** [review]
Personnaly i will say put it in every command locker, Loadout or not as its is a default item in my memory.

Swing sprite should be alligned better in my opinion.

And others things i suggest below.

File: Resources/Prototypes/Entities/Objects/Weapons/Melee/telescopic_baton.yml (hunk 1 above)
Comment (Remuchi) [reply Resources/Prototypes/Entities/Objects/Weapons/Melee/telescopic_baton.yml:28]: Have you ever been... beaten by this thing?
IT HURTS. IT HURTS ALOT.

Comment (FoxxoTrystan) [reply Resources/Prototypes/Entities/Objects/Weapons/Melee/telescopic_baton.yml:28]: I haven been and did beat peoples A LOT WITH IT.

But again that was on SS13/Paradise Station and damage there i think was near complete ZERO

**Review by Peptide90 — APPROVED** [review]

File: Content.Shared/StatusEffect/StatusEffectsSystem.cs [hunk 2]
```diff
@@ -123,6 +123,35 @@ public bool TryAddStatusEffect<T>(EntityUid uid, string key, TimeSpan time, bool
...
+        /// <param name="status">The status effects component to change, if you already have it.</param>
+        /// <returns>False if the effect could not be added or the component already exists, true otherwise.</returns>
+        public bool TryAddStatusEffect(EntityUid uid, string key, TimeSpan time, bool refresh, Component component,
+            StatusEffectsComponent? status = null)
+        {
+            if (!Resolve(uid, ref status, false))
+                return false;
+
+            if (!TryAddStatusEffect(uid, key, time, refresh, status))
+                return false;
```
Comment (VMSolidus) [inline Content.Shared/StatusEffect/StatusEffectsSystem.cs:143]:
```suggestion
            if (!Resolve(uid, ref status, false)
                || !TryAddStatusEffect(uid, key, time, refresh, status))
                return false;
```

**Review by VMSolidus — APPROVED** [review]

**Pull request merged**